scoreboard/app/Http/Controllers/RoomController.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Room;
+use Illuminate\Http\Request;
+
+class RoomController extends Controller
+{
+    public function show($room) {
+        $room = Room::where('name', $room)->firstOrFail();
+
+        return response()->json($room);
+    }
+
+    public function showTeams($room) {
+        $room = Room::where('name', $room)->firstOrFail();
+
+        $teams = $room->teams()->with('latestScore')->get();
+
+        return response()->json($teams);
+    }
+
+    public function store(Request $request)
+    {
+        $unique = false;
+
+        do {
```

Comment:
This can (and should) be a `while (!$unique) {`.

A do-while serves purpose, but here it's just bloat (the first step is executed anyway, as `$unique = false`.